lib/commands/device/startservice.ts

```diff
@@ -0,0 +1,155 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Flags, Args } from '@oclif/core';
+import Command from '../../command';
+import * as cf from '../../utils/common-flags';
+import { getBalenaSdk, getCliUx, stripIndent } from '../../utils/lazy';
+import type {
+	BalenaSDK,
+	DeviceWithServiceDetails,
+	CurrentServiceWithCommit,
+} from 'balena-sdk';
+
+export default class DeviceStartServiceCmd extends Command {
+	public static description = stripIndent`
+		Start containers on a device.
+
+		Start containers on a device.
+		If the --service flag is provided, then only those services' containers
+		will be started.
+
+		Multiple devices and services may be specified with a comma-separated list
+		of values (no spaces).
+		`;
+	public static examples = [
+		'$ balena device startservice 23c73a1 --service myService',
+		'$ balena device startservice 23c73a1 -s myService1,myService2',
+	];
+
+	public static args = {
+		uuid: Args.string({
+			description:
+				'comma-separated list (no blank spaces) of device UUIDs to start',
+			required: true,
+		}),
+	};
+
+	public static usage = 'device startservice <uuid>';
+
+	public static flags = {
+		service: Flags.string({
+			description:
+				'comma-separated list (no blank spaces) of service names to start',
+			char: 's',
+		}),
+		help: cf.help,
+	};
+
+	public static authenticated = true;
+
+	public async run() {
+		const { args: params, flags: options } = await this.parse(
+			DeviceStartServiceCmd,
+		);
+
+		const balena = getBalenaSdk();
+		const ux = getCliUx();
+
+		const deviceUuids = params.uuid.split(',');
+		const serviceNames = options.service?.split(',');
+
+		// Iterate sequentially through deviceUuids.
+		// We may later want to add a batching feature,
+		// so that n devices are processed in parallel
+		for (const uuid of deviceUuids) {
+			ux.action.start(`Starting services on device ${uuid}`);
+			if (serviceNames) {
+				await this.startServices(balena, uuid, serviceNames);
+			}
+			ux.action.stop();
+		}
+	}
+
+	async startServices(
+		balena: BalenaSDK,
+		deviceUuid: string,
+		serviceNames: string[],
+	) {
+		const { ExpectedError, instanceOf } = await import('../../errors');
+		const { getExpandedProp } = await import('../../utils/pine');
+
+		// Get device
+		let device: DeviceWithServiceDetails<CurrentServiceWithCommit>;
+		try {
+			device = await balena.models.device.getWithServiceDetails(deviceUuid, {
+				$expand: {
+					is_running__release: { $select: 'commit' },
+				},
+			});
+		} catch (e) {
+			const { BalenaDeviceNotFound } = await import('balena-errors');
+			if (instanceOf(e, BalenaDeviceNotFound)) {
+				throw new ExpectedError(`Device ${deviceUuid} not found.`);
+			} else {
+				throw e;
+			}
+		}
```

Comment:
Probably better we just wrap the call to `this.startServices()` in the trycatch. Also I don't think we need to explicitly catch the `BalenaDeviceNotFound` error, do we? An error of that instance should already have a relevant error message. Furthermore, we probably don't need the trycatch at all unless we want to output some specific text for the user (I expect you added it to manipulate the thrown error, right? Which as noted I don't think is necessary unless I'm missing something 🤔 )
```suggestion
		const device = await balena.models.device.getWithServiceDetails(deviceUuid, {
				$expand: {
					is_running__release: { $select: 'commit' },
				},
			});
```